libtransmission/peer-msgs.cc

```diff
@@ -565,6 +565,12 @@ class tr_peerMsgsImpl final : public tr_peerMsgs
         if (can_add_request_from_peer(req))
         {
             peer_requested_.emplace_back(req);
+
+            auto const now_sec = tr_time();
+            auto const now_msec = tr_time_msec();
+            while (fill_output_buffer(now_sec, now_msec) != 0U)
+            {
+            }
```

Comment:
_(nit)_

This is pretty similar to a block of code in `tr_peerMsgsImpl::pulse()`. Maybe it would make sense to extract this paragraph into its own private method so that both `pulse()` and `on_peer_made_request()` can call it?

Reply:
Also, idea LGTM in principle

Reply:
04e61a077d8d9f8eccf8a9a8e2c6b7d81977228a